src/Compilers/Core/Portable/DiagnosticAnalyzer/ArtifactGenerator.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// The base type for artifact generators that can programmatically produce additional files for a compilation.
+    /// </summary>
+    public abstract class ArtifactGenerator : DiagnosticAnalyzer
+    {
+        // By default artifact generators don't report diagnostics.  However, they are still allowed to if they run into
+        // any issues.
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray<DiagnosticDescriptor>.Empty;
+
+        public sealed override void Initialize(AnalysisContext context)
```

Comment:
sealed as we do not want the subclasses to control this at all.  they only override the various `GenerateArtifacts` overloads.